### PR TITLE
Check for null type trees in dependency extraction

### LINKED ISSUE #1655: 0.13.6 throws null pointer with macro paradise
clone this https://github.com/MasseGuillaume/sbt-example-paradise

git@github.com:MasseGuillaume/sbt-example-paradise.git

sbt "core/run"

### PR DESCRIPTION
**Pull request by Duhemm**
In some cases the dependency extraction may encounter a null `TypeTree`
(eg. arguments of macro annotations that are untyped). In such cases,
we simply ignore the node.

This solves all the crashes that I've been able to reproduce (https://github.com/sbt/sbt/issues/1655#issue-45323426, julianpeeters/avro-scala-macro-annotations#4, [Duhemm/macro-annotation](https://github.com/Duhemm/macro-annotation))

/cc @xeno-by @dragos

Fixes #1655

### CODE REVIEW

**Comment by typesafe-tools** [comment]
Can one of the admins verify this patch?

**Comment by jsuereth** [comment]
Ok, so we can't add the dependency because it's null.  Does this mean if you use a macro annotation that no dependency is ever recorded?

**Comment by dragos** [comment]
LGTM

**Comment by jsuereth** [comment]
Also, is there any way to get a test around this?

**Comment by Duhemm** [comment]
I just added a scripted test that uses a macro annotation. Would you prefer it to be introduced before as a pending test, and then mark it as fixed ?

**Comment by dragos** [comment]
I think it's fine like this (test + fix) in one go.

**Comment by jsuereth** [comment]
this looks great.  will merge once tests pass

**Comment by Duhemm** [comment]
Great, thanks !
Looks like something went wrong with the build: https://travis-ci.org/sbt/sbt/jobs/42855282#L6955
Could you please restart this job ?

File: sbt/src/sbt-test/source-dependencies/macro-annotation/project/Build.scala [hunk 1]
```diff
@@ -0,0 +1,45 @@
+import sbt._
+import Keys._
+
+object BuildSettings {
+  val paradiseVersion = "2.0.1"
+  val buildSettings = Defaults.defaultSettings ++ Seq(
+    version := "1.0.0",
+    scalacOptions ++= Seq(""),
```
Comment (jaceklaskowski) [inline sbt/src/sbt-test/source-dependencies/macro-annotation/project/Build.scala:8]: Why is this needed?!

File: sbt/src/sbt-test/source-dependencies/macro-annotation/project/Build.scala [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    scalaVersion := "2.11.4",
```
Comment (jaceklaskowski) [inline sbt/src/sbt-test/source-dependencies/macro-annotation/project/Build.scala:9]: Shouldn't it be `2.11.5` since the version's out?

File: sbt/src/sbt-test/source-dependencies/macro-annotation/project/Build.scala [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    settings = buildSettings ++ Seq(
+      run <<= run in Compile in core
+    )
+  ) aggregate(macros, core)
+
+  lazy val macros: Project = Project(
+    "macros",
+    file("macros"),
+    settings = buildSettings ++ Seq(
+      libraryDependencies <+= (scalaVersion)("org.scala-lang" % "scala-reflect" % _),
```
Comment (jaceklaskowski) [inline sbt/src/sbt-test/source-dependencies/macro-annotation/project/Build.scala:32]: I thought that `<+=` got deprecated (or at least new code should use it with care) since `scalaVersion.value` works and looks so much "better".
